src/btwa_api/app/db/firebase/firebase.py

```diff
@@ -0,0 +1,92 @@
+import csv
+from io import StringIO
+from threading import RLock
+
+import firebase_admin
+from firebase_admin import firestore, storage
+from starlette.requests import Request
+
+from ...config import FIREBASE_STORAGE_BUCKET
+
+MAX_IN_QUERY_LENGTH = 10
+ALLOWED_USER_STATUSES = {"unknown", "infected", "cured"}
+
+
+def get_users_batched(collection, ids):
```

Comment:
I was looking for some window/chunked function in `itertools` but this seemed quite elegant to me, slices can do it without any libraries :) Seems like an overkill to download another package for this to me.

Reply:
Strongly disagree, that's what libraries are for ;) I'll gladly refactor this if you don't have any objections.

Reply:
Indeed libraries are nice, when you have a sensible and integrated package management system, which Python sadly doesn't have :-( Feel free to refactor it, but as I said, there's a possibility that this code will go into the dumpster in the next few days.

Reply:
@bodik ok, what about this:

```python
from more_itertools import chunked
...

def get_users_batched(collection, ids):
    return [
        d.to_dict()
        for id_chunk in chunked(ids, MAX_IN_QUERY_LENGTH)
        for d in collection.where("buid", "in", id_chunk).get()
    ]
```

Given that this is mostly a nitpick, it's obviously your call if you feel it's a readability improvement (I do think it is).

But I'll admit I haven't run it nor added a test for it.

Reply:
Personally I hate double list comprehension, it twist my mind every time I see it. Since it's just a cosmetics I'd leave the original implementation as is for now.

Reply:
Not cosmetics imo, but it's up to you. If you find the double list expression hard to follow (understandable, as it's indeed a weird way of `flatmap`ping, albeit a standard, pythonic one), perhaps there's a middle ground where `chunked` is kept. I find the original implementation unreadable.